smssync/src/main/java/org/addhen/smssync/presentation/service/ServiceControl.java

```diff
@@ -207,13 +207,8 @@ public boolean isConnected() {
         return Utility.isConnected(mContext);
     }
 
-    public enum SchedulerInstance {
-        INSTANCE;
-
-        private Scheduler getScheduler(Context context, FileManager fileManager, Intent intent,
-                int requestCode) {
-            return new Scheduler(context, fileManager, intent, requestCode,
-                    PendingIntent.FLAG_UPDATE_CURRENT);
-        }
+    private Scheduler getScheduler(Intent intent, int requestCode) {
```

Comment:
oh :-) In that case let's maintain your `private Scheduler getScheduler(Intent intent, int requestCode) {` . And yes it makes more sense to enforce the singletonism ( Is this actually a word? Just curious ;-) ) in the Scheduler class itself.